src/client-functions/selector-builder/add-api.js

```diff
@@ -110,6 +113,38 @@ function addSnapshotPropertyShorthands (obj, getSelector, customDOMProperties) {
             }
         });
     });
+}
+
+function addCustomMethods (obj, getSelector, customMethods) {
+    var customMethodProps = customMethods ? Object.keys(customMethods) : [];
+
+    customMethodProps.forEach(prop => {
+        obj[prop] = (...args) => {
+            var callsite = getCallsite(prop);
+
+            return ClientFunctionResultPromise.fromFn(async () => {
+                var snapshot = await getSnapshot(getSelector, callsite);
```

Comment:
Why you need to obtain snapshot to execute client function, consequently making 2 client-server round trips?